provider/src/main/kotlin/org/gradle/kotlin/dsl/KotlinInitScript.kt

```diff
@@ -0,0 +1,384 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl
+
+import org.gradle.api.Action
+import org.gradle.api.PathValidation
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.ConfigurableFileTree
+import org.gradle.api.file.CopySpec
+import org.gradle.api.file.DeleteSpec
+import org.gradle.api.file.FileTree
+import org.gradle.api.invocation.Gradle
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
+import org.gradle.api.logging.LoggingManager
+import org.gradle.api.plugins.ObjectConfigurationAction
+import org.gradle.api.plugins.PluginAware
+import org.gradle.api.resources.ResourceHandler
+import org.gradle.api.tasks.WorkResult
+
+import org.gradle.kotlin.dsl.resolver.KotlinBuildScriptDependenciesResolver
+import org.gradle.kotlin.dsl.support.KotlinScriptHost
+import org.gradle.kotlin.dsl.support.serviceOf
+
+import org.gradle.process.ExecResult
+import org.gradle.process.ExecSpec
+import org.gradle.process.JavaExecSpec
+
+import java.io.File
+import java.net.URI
+
+import kotlin.script.extensions.SamWithReceiverAnnotations
+import kotlin.script.templates.ScriptTemplateDefinition
+
+
+/**
+ * Base class for Kotlin init scripts.
+ */
+@ScriptTemplateDefinition(
+    resolver = KotlinBuildScriptDependenciesResolver::class,
+    scriptFilePattern = ".+\\.init\\.gradle\\.kts")
+@SamWithReceiverAnnotations("org.gradle.api.HasImplicitReceiver")
+abstract class KotlinInitScript(
+    private val host: KotlinScriptHost,
+    gradle: Gradle) : Gradle by gradle {
+
+    /**
+     * Configures the classpath of the init script.
+     */
+    @Suppress("unused")
+    open fun initscript(@Suppress("unused_parameter") block: ScriptHandlerScope.() -> Unit) = Unit
+
+    /**
+     * Logger for settings. You can use this in your settings file to write log messages.
```

Comment:
:x: This should read *init scripts* instead of *settings*

Reply:
Right. Thanks.